test/PlayerActions/END_TURN.test.js

```diff
@@ -0,0 +1,126 @@
+const expect = require("chai").expect;
+const EventEmitter = require("events");
+const sinon = require("sinon");
+
+const mockUIFactory = require('../mocks/UI');
+
+const { GameState } = require("../../entities/GameState");
+const { createPlayer, setupMockDice } = require("../testutils");
+
+describe("PlayerActions -> END_TURN", () => {
+  let gameState;
+  let mockUI;
+  let eventBusEmitter;
+  let PLAYER_ACTIONS;
+  
+  beforeEach(() => {
+    gameState = new GameState();
+    eventBusEmitter = new EventEmitter();
+    mockUI = mockUIFactory();
+    
+    gameState.players = [createPlayer({ name: "player1" })];
+    PLAYER_ACTIONS = require("../../entities/PlayerActions")(eventBusEmitter, mockUI, gameState);
+  });
+  
+  afterEach(() => {
+    // Restore the default sandbox here
+    sinon.restore();
+  });
+
+  describe("endTurn", () => {
+    // The method under test
+    let methodUnderTest;
+    // Output Event from "rollDice"'s emit
+    const outputEvent = "START_TURN";
+    let outputSpy;
+
+    beforeEach(() => {
+      outputSpy = sinon.spy();
+      // Clear default listeners to prevent side-effects*
+      eventBusEmitter.removeAllListeners(outputEvent)
+      eventBusEmitter.on(outputEvent, outputSpy);
+
+      methodUnderTest = PLAYER_ACTIONS.END_TURN.execute;
+    });
+
+    it(`should emit ${ outputEvent } event`, () => {
+      methodUnderTest();
+
+      expect(outputSpy.callCount).to.equal(1, `${ outputEvent } event was not called`);
+    });
+    it(`should increment the turn counter`, () => {
+      const before = gameState.turn;
+      methodUnderTest();
+      const after = gameState.turn;
+
+      expect(after - before).to.equal(1, "Turn counter was not incremented");
+    });
+    it(`should make a call to the UI#endTurn`, () => {
+      const uiSpy = sinon.spy(mockUI, "endTurn");
+      methodUnderTest();
+
+      expect(uiSpy.callCount).to.equal(1);
+    });
+  });
+
+  describe("isAvailable", () => {
+    let methodUnderTest;
+    let mockRolls;
+
+    beforeEach(() => {
+      methodUnderTest = PLAYER_ACTIONS.END_TURN.isAvailable.bind(null, undefined, gameState);
```

Comment:
Is this the only way that we could write this?

Reply:
```js
methodUnderTest = () => PLAYER_ACTIONS.END_TURN.isAvailable(undefined, gameState);
```

Reply:
But that's dumb